pkg/globalnet/controllers/gateway_monitor.go

```diff
@@ -116,15 +116,15 @@ func NewGatewayMonitor(spec Specification, localCIDRs []string, config *watcher.
 func (g *gatewayMonitor) Start() error {
 	klog.Info("Starting GatewayMonitor to monitor the active Gateway node in the cluster.")
 
+	if err := g.createGlobalNetMarkingChain(); err != nil {
```

Comment:
Do we know what is causing the error? maybe we should understand this before we provide a specific fix?

It seems to me like some kind of race (I can see that we use 'cmd.Run ()' to create the ipt), maybe we need to figure out how to serialize it?

Reply:
> Do we know what is causing the error? maybe we should understand this before we provide a specific fix?

Actually there is no error. After we create the IPTable Chain it seems to take few milli-seconds to be sync'ed and in the meantime when we try to program a rule in that chain, its returning an error "No chain/target/match by that name." Subsequently after few milliseconds when the code retries to add the iptable rule in the chain, it succeeds.

> It seems to me like some kind of race, I can see that we set the iptables timeout to 5 seconds ?
The iptable timeout is for acquiring the xtables lock if there is any contention with some other process holding up the lock.

Reply:
Yeah, sorry, my bad, I meant what is causing the race (not the error)? 

If you think it's a critical fix for the release I guess we can continue with this PR, otherwise maybe it will be better to analyze this case first.

What do you think ?

Reply:
Well, this is not a critical fix, as the retry mechanism mitigates this issue. But the problem is that while looking at the logs, we will notice lot of errors which could be misleading. Also, we are not seeing this error on KIND setups but its only seen on OCP deployments.

Its good if we can analyse the issue in parallel and identify the root cause. But personally i feel that this is a harmless change as we are only ensuring that after creating a chain, its really sync'ed to the kernel before moving on.

Reply:
OK, maybe we can add the fix in a way that it will still be possible to identify traces of the original problem in the logs?

Reply:
I moved this specific fix to a separate PR - https://github.com/submariner-io/submariner/pull/1764